src/main/java/address/controller/PersonCardController.java

```diff
@@ -45,13 +45,43 @@ public PersonCardController(Person person) {
 
     @FXML
     public void initialize() {
-        firstName.setText(person.getFirstName());
-        lastName.setText(person.getLastName());
-        street.setText(person.getStreet());
-        postalCode.setText(person.getPostalCodeString());
-        city.setText(person.getCity());
-        birthday.setText(person.getBirthdayString());
-        contactGroups.setText(getContactGroupsString(person.getContactGroupsCopy()));
+        firstName.textProperty().bind(person.firstNameProperty());
+        lastName.textProperty().bind(person.lastNameProperty());
+        street.textProperty().bind(person.streetProperty());
+        postalCode.textProperty().bind(new StringBinding(){
+            {
+                bind(person.postalCodeProperty()); //Bind property at instance initializer
+            }
+
+            @Override
+            protected String computeValue() {
+                if(person.postalCodeProperty().get() == 0){
+                    return "";
+                }
+                return String.valueOf(person.postalCodeProperty().get());
```

Comment:
They don't have an asString method.

Get Outlook for Android<https://aka.ms/ghei36>



On Tue, May 24, 2016 at 3:19 AM -0700, "Nguyen Quoc Dat" <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/main/java/address/controller/PersonCardController.java<https://github.com/HubTurbo/addressbook/pull/116#discussion_r64365232>:

> -        birthday.setText(person.getBirthdayString());
> -        contactGroups.setText(getContactGroupsString(person.getContactGroupsCopy()));
> +        firstName.textProperty().bind(person.firstNameProperty());
> +        lastName.textProperty().bind(person.lastNameProperty());
> +        street.textProperty().bind(person.streetProperty());
> +        postalCode.textProperty().bind(new StringBinding(){
> +            {
> +                bind(person.postalCodeProperty()); //Bind property at instance initializer
> +            }
> +
> +            @Override
> +            protected String computeValue() {
> +                if(person.postalCodeProperty().get() == 0){
> +                    return "";
> +                }
> +                return String.valueOf(person.postalCodeProperty().get());


What about the other properties below that don't need to deal with special cases?

-
You are receiving this because you modified the open/close state.
Reply to this email directly or view it on GitHub<https://github.com/HubTurbo/addressbook/pull/116/files/b8b848d25598506a2938bd00e487a44220b57b60#r64365232>